homeassistant/components/dwd_weather_warnings/config_flow.py

```diff
@@ -26,27 +28,58 @@ async def async_step_user(
         errors: dict = {}
 
         if user_input is not None:
-            region_identifier = user_input[CONF_REGION_IDENTIFIER]
+            valid_options = (CONF_REGION_IDENTIFIER, CONF_GPS_TRACKER)
 
-            # Validate region identifier using the API
-            if not await self.hass.async_add_executor_job(
-                DwdWeatherWarningsAPI, region_identifier
-            ):
-                errors["base"] = "invalid_identifier"
+            # Check, if either CONF_REGION_IDENTIFIER or CONF_GPS_TRACKER has been set.
+            if all(k not in user_input for k in valid_options):
+                errors["base"] = "no_identifier"
+            elif all(k in user_input for k in valid_options):
+                errors["base"] = "ambiguous_identifier"
+            elif CONF_REGION_IDENTIFIER in user_input:
+                # Validate region identifier using the API
+                identifier = user_input[CONF_REGION_IDENTIFIER]
 
-            if not errors:
-                # Set the unique ID for this config entry.
-                await self.async_set_unique_id(region_identifier)
-                self._abort_if_unique_id_configured()
+                if not await self.hass.async_add_executor_job(
+                    DwdWeatherWarningsAPI, identifier
+                ):
+                    errors["base"] = "invalid_identifier"
 
-                return self.async_create_entry(title=region_identifier, data=user_input)
+                if not errors:
+                    # Set the unique ID for this config entry.
+                    await self.async_set_unique_id(identifier)
+                    self._abort_if_unique_id_configured()
+
+                    return self.async_create_entry(title=identifier, data=user_input)
+            elif CONF_GPS_TRACKER in user_input:
+                # Validate position using the API
+                device_tracker = user_input[CONF_GPS_TRACKER]
+                position = get_position_data(self.hass, device_tracker)
+
+                if not await self.hass.async_add_executor_job(
+                    DwdWeatherWarningsAPI, position
+                ):
+                    errors["base"] = "invalid_identifier"
+
+                # Position is valid here, because the API call was successful.
+                if not errors and position is not None:
+                    # Set the unique ID for this config entry.
+                    await self.async_set_unique_id(f"{position[0]}-{position[1]}")
```

Comment:
This is not a good source of unique_id, how about using the entity registry ID if the device tracker is in the entity registry instead?

Reply:
Is this the ID you suggested? (not really familiar with the entity registry sorry)

```python
registry = er.async_get(self.hass)
entity_entry = registry.async_get(device_tracker)
await self.async_set_unique_id(entity_entry.id)
```

Reply:
Yes, that's the one 👍

Reply:
Updated in commit 715d339bf96fa18895fc7b0656c2abd5e49b82d5